### PR TITLE
add_cache_pollution: Add DATA cache pollution

### PR DESCRIPTION
**Pull request by AbhayPatil**

### CODE REVIEW

File: activity.cc [hunk 1]
```diff
@@ -65,4 +69,61 @@ absl::Status WasteCpu::ValidateConfig(ActivityConfig& ac) {
...
+
+  std::random_device rd;
+  rand_gen_ = std::mt19937(rd());
+}
+
+void PolluteDataCache::DoActivity() {
+  iteration_count_++;
+  int64_t sum = 0;
+  for (int i = 0; i < array_reads_per_iteration_; i++) {
+    int index = random_index_(rand_gen_);
```
Comment (danmanj) [inline activity.cc:113]: this is going to spend more time generating random numbers than polluting the cache,

File: activity.cc [hunk 2]
```diff
@@ -65,4 +69,66 @@ absl::Status WasteCpu::ValidateConfig(ActivityConfig& ac) {
...
+
+  data_array_.resize(array_size);
+  for (int i=0; i<array_size; i++) {
+    data_array_[i] = i;
+  }
+
+  random_index_ = std::uniform_int_distribution<>(0, array_size - 1);
+  array_reads_per_iteration_ =
+      config->pollute_data_cache_config.array_reads_per_iteration;
+  activity_config_name_ = config->activity_config_name;
```
Comment (danmanj) [inline activity.cc:104]: this seems otherwise to be unused

Comment (AbhayPatil) [reply activity.cc:104]: Removed it

File: activity.cc [hunk 3]
```diff
@@ -65,4 +68,65 @@ absl::Status WasteCpu::ValidateConfig(ActivityConfig& ac) {
...
+
+  std::random_device rd;
+  rand_gen_ = std::mt19937(rd());
+}
+
+void PolluteDataCache::DoActivity() {
+  iteration_count_++;
+  int64_t sum = 0;
+  for (int i = 0; i < array_reads_per_iteration_; i++) {
+    // 'rand_gen_' is Mersenne Twister generator.
```
Comment (danmanj) [inline activity.cc:113]: a better name for the variable would make this comment unnecessary, mersene_twister_prng_ for instance

**Review by danmanj — APPROVED** [review]

File: activity.cc (hunk 3 above)
Comment (AbhayPatil) [reply activity.cc:113]: Agreed. Done

**Pull request merged**